dpa_adapt/grouped/_convert.py

```
@@ -0,0 +1,369 @@
+                sysdir, skipped=True, reason=f"{set_dir.name} has no coord/real_atom_types"
+            )
+        per_set.append((set_dir, nframes, natoms, real_types))
+
+    total = sum(n for _, n, _, _ in per_set)
+    group_ids = _assign_group_ids(group_by, per_set, property_name)
+
+    result = GroupMarkerResult(
+        sysdir, n_frames=total, n_groups=len(np.unique(group_ids))
+    )
+    offset = 0
+    for set_dir, nframes, natoms, real_types in per_set:
```

Ruff (0.15.20)

[warning] Loop control variable `natoms` not used within loop body

(B007)